hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -260,13 +304,19 @@ def get_metadata_objects_from_match(
     def get_index(self, signal_type: t.Type[SignalType]) -> SignalTypeIndex:
```

Comment:
Asked elsewhere too, but this seems like THE place where HMA is figuring out dynamically which index to use. mappings.py seems more static. 

Do you see this as a better place to plop the logic?

Reply:
sure!